crates/bevy_ecs/src/system/query.rs

```diff
@@ -973,8 +962,6 @@ where
     /// ```
     #[inline]
     pub fn is_empty(&self) -> bool {
-        // TODO: This code can be replaced with `self.iter().next().is_none()` if/when
-        // we sort out how to convert "write" queries to "read" queries.
         self.state
```

Comment:
We should replace this with the code in the comment now that we've figured out that conversion. I expect it will be both cleaner and marginally faster.

Reply:
`QueryState::is_empty` existed and got updated so I left this as a pass through to it to avoid changing any `pub fn`

Also the implementation isn't quite what is described in the comment as the actual implementation uses a custom `Fetch` to avoid touching any of the data much like the custom `is_empty` did (`NopFetch`)

Reply:
Sounds good to me.